src/FinalMileCarrier/FinalMileCarrier.php

```diff
@@ -0,0 +1,154 @@
+        'MX' => ['name' => self::MEXICO, 'url' => 'https://www.correosdemexico.gob.mx/SSLServicios/SeguimientoEnvio/seguimientoportal2.aspx?guia=%s'],
+        'NL' => ['name' => self::POSTNL, 'url' => 'https://postnl.post/'],
+        'NZ' => ['name' => self::NEW_ZEALAND_POST, 'url' => 'https://www.nzpost.co.nz/tools/tracking'],
+        'NG' => ['name' => self::NIGERIAN_POSTAL_SERVICE, 'url' => 'https://www.nipost.gov.ng/Track_Trace'],
+        'NO' => ['name' => self::POSTEN_NORGE, 'url' => 'https://sporing.posten.no/sporing/%s?lang=no'],
+        'PH' => ['name' => self::PHLPOST, 'url' => 'https://www.phlpost.gov.ph/'],
+        'PL' => ['name' => self::POCZTA_POLSKA, 'url' => 'https://emonitoring.poczta-polska.pl/'],
+        'PT' => ['name' => self::CTT, 'url' => 'https://www.ctt.pt/feapl_2/app/open/objectSearch/objectSearch.jspx?lang=def'],
+        'RO' => ['name' => self::POȘTA_ROMÂNĂ, 'url' => 'https://www.posta-romana.ro/track-trace.html'],
+        'RU' => ['name' => self::RUSSIA_POST, 'url' => 'https://www.pochta.ru/tracking#%s'],
+        'SG' => ['name' => self::SINGAPORE_POST, 'url' => 'https://www.singpost.com/send-receive/receive-mail-parcel'],
+        'SK' => ['name' => self::SLOVENSKÁ_POŠTA, 'url' => 'https://tandt.posta.sk/'],
+        'SI' => ['name' => self::POŠTA_SLOVENIJE, 'url' => 'https://sledenje.posta.si/'],
+        'ZA' => ['name' => self::SOUTH_AFRICAN_POST_OFFICE, 'url' => 'http://globaltracktrace.ptc.post/gtt.web/'],
+        'ES' => ['name' => self::CORREOS, 'url' => 'https://www.correos.es/es/es/herramientas/localizador/envios/detalle?tracking-number=%s'],
+        'SE' => ['name' => self::POSTNORD, 'url' => 'https://www.postnord.se/vara-verktyg/spara-brev-paket-och-pall?shipmentId=%s'],
+        'CH' => ['name' => self::SWISS_POST, 'url' => 'https://service.post.ch/ekp-web/ui/list'],
+        'TW' => ['name' => self::CHUNGHWA_POST, 'url' => 'https://postserv.post.gov.tw/pstmail/main_mail.html?targetTxn=%s'],
+        'TH' => ['name' => self::THAILAND_POST, 'url' => 'https://track.thailandpost.co.th/'],
+        'TR' => ['name' => self::PTT, 'url' => 'https://gonderitakip.ptt.gov.tr/Track/Result'],
+        'UK' => ['name' => self::ROYAL_MAIL, 'url' => 'https://www.royalmail.com/track-your-item#/tracking-results/%s'],
+        'UY' => ['name' => self::CORREO_URUGUAYO, 'url' => 'https://www.correo.com.uy/seguimientodeenvios'],
+        'US' => ['name' => self::USPS, 'url' => 'https://tools.usps.com/go/TrackConfirmAction?tRef=fullpage&tLc=2&text28777=&tLabels=%s%2C&tABt=false'],
+    ];
+
+    /**
+     * @throws CarrierNotSupportedException
+     */
+    public function __construct(private string $countryCode, private string $trackingCode)
+    {
+        $finalMileCarrier = in_array($this->countryCode, array_keys(self::FINAL_MILE_CARRIERS)) ? self::FINAL_MILE_CARRIERS[$this->countryCode] : throw new CarrierNotSupportedException($this->countryCode);
```

Comment:
I get what you are doing here but I personally do prefer an if statement instead of the one-liner. There are two reasons I can think of and they are:
- readibility (now the throw is at the end)
- $finalMileCarrier = throw ... - although obvious what will it do, the `throw` statement does not return a value